src/index.ts

```diff
@@ -99,6 +100,68 @@ export const jsonSchemaTransform = createJsonSchemaTransform({
   skipList: defaultSkipList,
 });
 
+export const createJsonSchemaTransformObject =
+  ({ schemas: zodSchemas }: { schemas: Record<string, z.ZodTypeAny> }) =>
+  (
+    input:
+      | { swaggerObject: Partial<OpenAPIV2.Document> }
+      | { openapiObject: Partial<OpenAPIV3.Document | OpenAPIV3_1.Document> },
+  ) => {
+    if ('swaggerObject' in input) {
+      console.warn('This package currently does not support component references for Swagger 2.0');
+      return input.swaggerObject;
+    }
+
+    const schemas: Record<string, OpenAPIV3.SchemaObject | OpenAPIV3_1.SchemaObject> = {};
+    for (const key in zodSchemas)
+      schemas[key] = zodToJsonSchema(zodSchemas[key], zodToJsonSchemaOptions);
+
+    const document = {
+      ...input.openapiObject,
+      components: {
+        ...input.openapiObject.components,
+        schemas: {
+          ...input.openapiObject.components?.schemas,
+          ...schemas,
+        },
+      },
+    };
+
+    const componentMapVK = new Map<string, string>();
+    Object.entries(schemas).forEach(([key, value]) =>
+      componentMapVK.set(JSON.stringify(value), key),
+    );
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    function componentReplacer(this: any, key: string, value: any) {
+      if (typeof value !== 'object') return value;
+
+      // Check if the parent is the schemas object, if so, return the value as is
+      if (this === document.components.schemas) return value;
+
+      const stringifiedValue = JSON.stringify(value);
```

Comment:
can you use ``fast-json-stringify` instead?

Reply:
I don't think it has a replacer function options however.

Reply:
not suggesting it for all cases, just where we do straightforward stringification